src/nvim/eval.c

```diff
@@ -24033,6 +24050,7 @@ static inline TerminalJobData *common_job_init(char **argv,
   }
   Process *proc = (Process *)&data->proc;
   proc->argv = argv;
+  proc->quote_argv = quote_argv;
```

Comment:
Would not it be better to be
proc->quote_argv = *p_sxq == NUL ? false : quote_argv;